sdk/communication/Azure.Communication.Rooms/src/RoomsClient.cs

```diff
@@ -0,0 +1,511 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Azure.Communication.Pipeline;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication.Rooms
+{
+    /// <summary>
+    /// The Azure Communication Services Rooms client.
+    /// </summary>
+    public class RoomsClient
+    {
+        private readonly ClientDiagnostics _clientDiagnostics;
+
+        internal RoomsRestClient RoomsServiceClient { get; }
+
+        #region public constructors - all arguments need null check
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RoomsClient"/> class.
+        /// </summary>
+        /// <param name="connectionString"></param>
+        public RoomsClient(string connectionString)
+            : this(
+                ConnectionString.Parse(Argument.CheckNotNullOrEmpty(connectionString, nameof(connectionString))),
+                new RoomsClientOptions())
```

Comment:
Nit - consider minimizing logic and just doing `this(connectionString, null)`.

Reply:
The initializations are pretty standard formatted with other communication services.
Please refer to email client.
Thanks.